src/scripts/Update.ts

```diff
@@ -53,311 +239,86 @@ class Update implements Saveable {
-                delete playerData._itemList.Time_stone;
-
-                // Update player data
-                this.setPlayerData(playerData);
-            } catch (ಠ_ಠ) {
-                console.error('[update] v0.5.0 - Couldn\'t update item list data..', ಠ_ಠ);
-            }
+        if (!playerData || !saveData || !settingsData) {
+            return;
         }
 
-        if (this.isOlderVersion(this.saveVersion, '0.5.2')) {
-            try {
-                // Calculate hatched amount (we can't calculate the shiny hatches though)
-                const pokemonHatched = {};
-                saveData.party.caughtPokemon.forEach(p => pokemonHatched[p.id] = p.attackBonus / 25);
-                // Rename from the old statistic name, add our new statistics
-                saveData.statistics = {
-                    ...saveData.statistics,
-                    totalBerriesHarvested: saveData.statistics.berriesHarvested.reduce((sum, b) => sum + b, 0) || 0,
-                    totalShardsGained: saveData.statistics.totalShards.reduce((sum, b) => sum + b, 0) || 0,
-                    shardsGained: saveData.statistics.totalShards || 0,
-                    pokemonHatched,
-                };
-
-                // Update save data
-                this.setSaveData(saveData);
-            } catch (ಠ_ಠ) {
-                console.error('[update] v0.5.2 - Couldn\'t update player statistics..', ಠ_ಠ);
-            }
-            try {
-                // If the player has the Soul Badge already
-                // Not using game constants incase the badge value isn't 5 in the future
-                if (saveData.badgeCase[5]) {
-                    saveData.quests.questLines.push({state: 1, name: 'Mining Expedition', quest: 0});
+        const updateResult = Object.entries(this.updateSteps)
+            .filter(([version]) => this.isOlderVersion(this.saveVersion, version))
+            .sort(([a], [b]) => this.isNewerVersion(a, b) ? 1 : -1)
+            .reduce((beforeUpdate, [version, callback]) => {
+                // The parse(stringify(...)) will deep-clone the object so
+                // that it isn't mutated, and we can log if something fails
+                const updateData = JSON.parse(JSON.stringify(beforeUpdate));
+                try {
+                    console.info(`Applying update v${version}`);
+                    callback(updateData);
+                    updateData.saveData.update.version = version;
+                    return updateData;
+                } catch (e) {
+                    console.error(`Caught error while applying update v${version}`, e, { beforeUpdate, updateData });
+                    Notifier.notify({
+                        title: `Failed to update to v${this.version}!`,
+                        message: 'Please check the console for errors, and report them on our Discord.',
+                        type: NotificationConstants.NotificationOption.danger,
+                        timeout: 6e4,
```

Comment:
Maybe we should extend the timeout of this notification.
```suggestion
                        timeout: GameConstants.HOUR,
```

Reply:
I set it to a full day, just in case someone walks away from the computer mid-update. If they take longer, the notification will appear, but most people will refresh the page and get the error again